packages/expo-dev-launcher/bundle/functions/getLocalPackagersAsync.ts

```diff
@@ -0,0 +1,38 @@
+import { Platform } from 'react-native';
+
+const baseAddress = Platform.select({
+  ios: 'http://localhost',
+  android: 'http://10.0.0.2',
+});
+const statusPage = 'status';
+const portsToCheck = [8081, 8082, 19000, 19001, 19002, 19003, 19004, 19005];
+
+export type Packager = {
+  description: string;
+  url: string;
+  source: string;
+  hideImage: boolean;
+};
+
+export async function getLocalPackagersAsync(): Promise<Packager[]> {
+  const onlinePackagers: Packager[] = [];
+
+  await Promise.all(
+    portsToCheck.map(async (port) => {
+      try {
+        const address = `${baseAddress}:${port}`;
+        const { status } = await fetch(`${address}/${statusPage}`);
```

Comment:
if the fetch call fails for whatever reason it's caught and ignored - Im not sure if that answers your question or not - I haven't noticed any hanging in personal use but it might be worth looking at it a bit harder if you think this might be bug prone!